File: Loom.Core/Models/ToolDefinition.vb
Namespace Models
    ''' <summary>
    ''' Defines tool content
    ''' </summary>
    Public Class ToolDefinition
        Public Property ToolName As String
        Public Property Description As String
        Public Property Parameters As List(Of ToolParameters)
        Public Property Required As List(Of String)
        Public Property Strictness As Boolean?
    End Class


End Namespace

File: implementations/csharp/src/Loom.Core/Models/ToolDefinition.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    /// <summary>
    /// Defines tool content
    /// </summary>
    public class ToolDefinition
    {
        public string ToolName { get; set; }
        public string Description   { get; set; }
        public bool? Strictness { get; set; }
        public List<string> Required { get; set; }
        public List<ToolParameters> Parameters { get; set; }
    }
}
